SMP: Add connect jetpack into the split button

sejas opened the pull request:
#### Proposed Changes

* Refactor Add New Site to be a Split button
* Add "Connect Jetpack Site" as a popover menu item inside the split button.
* Send `cta_from` query param to be tracked in the event `calypso_jpc_url_view`
* SplitButton: accept `href` new prop
* SplitButton accept `whiteSeparator` new prop

#### Screenshots

| Before | After |
|---|---|
| <img width="267" alt="67646-before" src="https://user-images.githubusercontent.com/779993/189676529-7b5ef304-14ff-4067-866e-bc4fb6b903af.png">   |  <img width="273" alt="Screenshot 2022-09-13 at 15 21 22" src="https://user-images.githubusercontent.com/779993/189926471-e66ca00a-3afd-43ec-8f27-7832888dba16.png"> |


#### Testing Instructions

<!--
Add as many details as possible to help others reproduce the issue and test the fix.
"Before / After" screenshots can also be very helpful when the change is visual.
-->

* Go to `/sites`
* Observe the `Add new site` on the top-right corner is a split button
* Click on it
* Observe you go to the add new site flow as you would do before this PR.
* Click on `< Back to Sites` to go back to `/sites`
* On the top-right corner, click on the small arrow of the split button.
* Observe that a menu appears
* Click on `Connect Jetpack site`
* Observe that you are redirected to `/jetpack/connect`

#### Pre-merge Checklist

<!--
Complete applicable items on this checklist **before** merging into trunk. Inapplicable items can be left unchecked.

Both the PR author and reviewer are responsible for ensuring the checklist is completed.
-->

- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [x] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [x] Have you checked for TypeScript, React or other console errors?
- [x] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?

<!--
Link a related issue to this PR. If the PR does not immediately resolve the issue,
for example, it requires a separate deployment to production, avoid
using the "fixes" keyword and instead attach the [Status] Fix Inbound label to
the linked issue.
-->

- Closes #67646

sejas committed:
SMP: update split button to accept href as prop

sejas committed:
SMP: add jetpack site into the split button

sejas commented:
I've added a couple of questions in the issue: https://github.com/Automattic/wp-calypso/issues/67646#issuecomment-1244039418

sejas committed:
Components: update SplitButton dev docs

sejas commented:
- Would be `Connect a Jetpack site` a better sentence?

- Should we add a "back to sites" button when we navigate to the Jetpack page from the SMP, similar to the Add new site page?

**Current Jetpack Flow**
<img width="1228" alt="Screenshot 2022-09-12 at 15 01 44" src="https://user-images.githubusercontent.com/779993/189673882-aa334ba8-0f1d-4dea-908d-912149b8caad.png">

**Current Add new site Flow**
<img width="1260" alt="189674005-538859ae-ac3d-4653-a04b-d3cb8819f60b" src="https://user-images.githubusercontent.com/779993/189714833-790bbd74-3c75-49fd-bfb0-fa7c28f3a1ff.png">

vindl commented:
Thanks for starting this @sejas! 

I think we should add a separator between the `Add New Site` and the chevron icon, to make it clear that the default action is different. Similar to this example:

<img width="261" alt="Screenshot 2022-09-12 at 22 44 33" src="https://user-images.githubusercontent.com/1182160/189755297-54e6d07d-63ce-4135-9d27-f48989ebaa75.png">

The width of the dropdown button could be reduced too, and we should use a Jetpack icon instead of the down arrow.

sejas commented:
About the separator, I'm currently using the split button on calypso. But I'm happy to add exceptions or new props to improve the design 👌 .
https://wpcalypso.wordpress.com/devdocs/design/split-button

wojtekn commented on client/components/split-button/index.jsx:
```diff
@@ -91,15 +92,19 @@ class SplitButton extends PureComponent {
 					disabled={ disabled || disableMain }
 					className="split-button__main"
+					onClick={ onClick }
+					href={ this.props.href }
```
I see that in other places in Calypso, we are using the `onClick` callback to handle actions related to the navigation, e.g.:

https://github.com/Automattic/wp-calypso/blob/trunk/client/my-sites/activity/activity-log-tasklist/index.jsx#L349

Would it make sense to follow the same way instead of adding support for the `href` to the component? That way, it may be easier to add more logic around that e.g., we can easily add tracking to that action.

wojtekn replied:
Another thought, I checked how we do it in `PopoverMenuItem` and in some cases we use a `href` for the standard lnk behavior and at the same time `onClick` callback to bump a stat.

sejas replied:
Yes, and I think it adds an extra complexity that is not necessary.
The `SplitButton` uses the Automattic Button component under the hood, which already supports the `href` prop, and it renders a link instead of a button which is more semantic, and it also tracks all the navigations automatically.

sejas committed:
Merge branch 'trunk' of github.com:Automattic/wp-calypso into add/smp-connect-jetpack-split-button

sejas committed:
Components: add white separator prop and style to SplitButton

sejas committed:
SMP: add white separator to the add new site split button

sejas committed:
SMP: replace connect jetpack site icon to use the logo

sejas committed:
SMP: small change on english sentence 'Connect *a* Jetpack site'

sejas committed:
Components: update SplitButton documentation to fix eslint a11y errors

sejas commented:
I've added a separator and replaced the jetpack logo:
![Screen Capture on 2022-09-13 at 15-33-33](https://user-images.githubusercontent.com/779993/189929415-5a4dcb98-3308-4fe7-9961-5081fdb3312e.gif)

sejas committed:
SMP: add cta_from and cta_id to jetpack connect to track events

wojtekn commented on client/components/split-button/docs/example.jsx:
```diff
@@ -32,15 +32,16 @@ class SplitButtonExample extends PureComponent {
 							[
 								{ label: 'Split Button', icon: 'history' },
 								{ label: 'Split Button', primary: true },
+								{ label: 'Split Button with whtie separator', primary: true, separatorWhite: true },
```
We need to fix the typo here. Also, we should fix the separator prop name as it does not match implementation - probably a refactoring issue:

```suggestion
								{ label: 'Split Button with white separator', primary: true, whiteSeparator: true },
```

sejas replied:
Thank you!, good catch!! 💪

wojtekn reviewed (commented):
@sejas looks good with the separator and the Jetpack icon. 

I see one more thing - would it be wiser to call the new prop just the `separator'? That way, if we want to change the color, we could adjust the CSS and change the color without changing the JS code and without leaving inconsistency.

sejas committed:
Components: refactor SplitButton example doc

sejas committed:
Components: update SplitButton documentation to fix eslint a11y errors

sejas committed:
Components: update SplitButton docs example

zaguiini reviewed (approved):
Very very good!

SaxonF commented:
Does "Add Jetpack to a self-hosted site" make more sense? 🤔 More context [here](pbNhbs-1Yo-p2#comment-9228)

sejas committed:
Component: change SplitButton white separator color

sejas committed:
SMP: change connect jetpack wording

sejas committed:
Merge branch 'trunk' of github.com:Automattic/wp-calypso into add/smp-connect-jetpack-split-button

sejas commented:
> Does "Add Jetpack to a self-hosted site" make more sense? 🤔 More context pbNhbs-1Yo-p2#comment-9228

@SaxonF, It does make more sense. I've changed the wording and the separator color. And this is how it looks

![connect-jetpack](https://user-images.githubusercontent.com/779993/190122392-8ef040e4-ba5b-4b43-a7b0-c5732e4ee825.gif)

a8ci18n commented:
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/7527535

Thank you @sejas for including a screenshot in the description! This is really helpful for our translators.

a8ci18n commented:
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/7527535).
